exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/ScanBatch.java

```diff
@@ -196,8 +196,8 @@ public IterOutcome next() {
             releaseAssets();
             done = true;  // have any future call to next() return NONE
 
-            if (mutator.isNewSchema()) {
-              // This last reader has a new schema (e.g., we have a zero-row
+            if (mutator.isNewTopLevelSchema()) {
```

Comment:
I don't see why this condition is better than isNewSchema. Looks to me that any time this is true, isNewSchema will also be true. Why don't we want to check the condition that the deeperSchema might have changed?